hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/ipc/CallerContext.java

```diff
@@ -114,6 +115,12 @@ public String toString() {
   /** The caller context builder. */
   public static final class Builder {
     private static final String KEY_VALUE_SEPARATOR = ":";
+    /**
+     * The illegal separators include '\t', '\n', '=', etc.
+     * User should not set illegal separator.
+     */
+    private static final List<String> ILLEGAL_SEPARATORS =
+        Arrays.asList("\t","\n","=");
```

Comment:
We should use `Collections.unmodifiableList` to provide an unmodifiable view.

Reply:
I would do a set: Collections.unmodifiableSet()

Reply:
Ok. Done